pkg/util/provider/machinecontroller/machine.go

```diff
@@ -577,23 +542,57 @@ func (c *controller) triggerCreationFlow(ctx context.Context, createMachineReque
 			TimeoutActive:  true,
 			LastUpdateTime: metav1.Now(),
 		}
-
 		_, err := c.controlMachineClient.Machines(clone.Namespace).UpdateStatus(ctx, clone, metav1.UpdateOptions{})
 		if err != nil {
 			klog.Warningf("Machine/status UPDATE failed for %q. Retrying, error: %s", machine.Name, err)
 		} else {
 			klog.V(2).Infof("Machine/status UPDATE for %q during creation", machine.Name)
-
 			// Return error even when machine object is updated
 			err = fmt.Errorf("Machine creation in process. Machine/Status UPDATE successful")
 		}
-
 		return machineutils.ShortRetry, err
 	}
-
 	return machineutils.LongRetry, nil
 }
 
+func (c *controller) updateLabelsAndInitializeMachine(ctx context.Context, createMachineRequest *driver.CreateMachineRequest, nodeName, providerID string, shouldInitializeMachine bool) (retryPeriod machineutils.RetryPeriod, err error) {
+	_, machineNodeLabelPresent := createMachineRequest.Machine.Labels[v1alpha1.NodeLabelKey]
+	_, machinePriorityAnnotationPresent := createMachineRequest.Machine.Annotations[machineutils.MachinePriority]
+	clone := createMachineRequest.Machine.DeepCopy()
+	if !machineNodeLabelPresent || !machinePriorityAnnotationPresent || createMachineRequest.Machine.Spec.ProviderID == "" {
+		if clone.Labels == nil {
+			clone.Labels = make(map[string]string)
+		}
+		clone.Labels[v1alpha1.NodeLabelKey] = nodeName
+		if clone.Annotations == nil {
+			clone.Annotations = make(map[string]string)
+		}
+		if clone.Annotations[machineutils.MachinePriority] == "" {
+			clone.Annotations[machineutils.MachinePriority] = "3"
+		}
+		clone.Spec.ProviderID = providerID
+		var updatedMachine *v1alpha1.Machine
+		updatedMachine, err = c.controlMachineClient.Machines(clone.Namespace).Update(ctx, clone, metav1.UpdateOptions{})
```

Comment:
I am not too sure if the return value which you have now initialized `updatedMachine` to contains the revised version/generation. Can you double check this?

Reply:
We returned a custom error from `InitializeMachine`, and saw that the status update went through.
```
E0912 09:40:01.094021   69341 machine.go:619] Error occurred while initializing VM instance for machine "shoot--i749592--it-worker-f73ns-z1-56778-ws5qx": machine codes error: code = [Uninitialized] message = [TESTING]
I0912 09:40:01.262395   69341 machine_util.go:565] Machine/status UPDATE for "shoot--i749592--it-worker-f73ns-z1-56778-ws5qx"
```

So we think that the update call is returning the updated Machine object.